src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -109,36 +110,41 @@ export abstract class BaseEslintEngine implements RuleEngine {
 		return await this.config.getTargetPatterns(this.strategy.getEngine());
 	}
 
-	async getCatalog(): Promise<Catalog> {
-		const categoryMap: Map<string, RuleGroup> = new Map();
-		const catalog: Catalog = {rulesets: [], categories: [], rules: []};
-		const rules: Rule[] = [];
-
-		// Get all rules supported by eslint
-		const cli = this.baseDependencies.createCLIEngine(this.strategy.getCatalogConfig());
-		const allRules = cli.getRules();
-
-		// Add eslint rules to catalog
-		allRules.forEach((esRule: ESRule, key: string) => {
-			const docs = esRule.meta.docs;
-
-			const rule = this.processRule(key, docs);
-			if (rule) {
-				// Add only rules supported by the engine implementation
-				rules.push(rule);
-				const categoryName = docs.category;
-				let category = categoryMap.get(categoryName);
-				if (!category) {
-					category = { name: categoryName, engine: this.getName(), paths: [] };
-					categoryMap.set(categoryName, category);
+	getCatalog(): Promise<Catalog> {
+		if (!this.catalog) {
```

Comment:
Caching this value because it is used by the DefaultRuleManager and BaseEslintEngines.